### ISSUE TITLE
When you enable both user and registration email verification, the user gets two emails

### ISSUE THREAD

**Issue by mooreds** [FusionAuth/fusionauth-issues#842]
### Description 
When you enable both user and registration email verification, the user gets two emails to verify, and only one of them has the correct code. The other one says "Sorry. The request contains an invalid or expired verification Id. You may need to request another verification to be sent."

### Affects versions

1.18.7

### Steps to reproduce
1. Make sure email verification is enabled under "Tenants / Email" and "Applications / Your application / Registration". Also make sure self service registration is enabled.
1. Configure your mail server in "Tenants / Email"
1. Create a user and a registration at the same time. I've used the API and a registration form.
1. The user will get two emails.

If you click on the latter one (I copy and paste it to an incognito window), you'll see the error.

From looking at the user source and the registration data in the admin UI, it looks like the user object gets 'verified' and the registration object does not.

I used the same email template for both of these.

### Expected behavior

I'm not sure. I think that at the least we shouldn't send two emails, one of which has an invalid code. We could document this behavior. We could have the registration take precedence when doing both. I'm not sure what the best path is.

### Platform

n/a

### Additional context

Came out of testing for this forum post: https://fusionauth.io/community/forum/topic/352/user-email-verified-webhook-not-firing/4

**Comment by robotdan** [comment]
I'll have to look at this, it is likely working as designed. 

We used to have a similar issue when using setup password and email verification. These both get sent during the user create, so we now collapse these two events into one using the Setup Password email which will also verify the email when complete. 

We could review doing something like that for this item, however I don't know if there is a use case to enable email verification and registration verification if you're using self service registration (which creates the user and registerers at the same time).

**Comment by cpoyatos1** [comment]
+1

**Comment by robotdan** [comment]
As far as I can tell this is working as designed. These are two discrete workflows, and I don't see how we could communicate to the user why we only sent one of them. 

The likely that anyone is using both email verification and registration verification low. 

After collapsing the setup password and email verification, we had to add back a policy to opt into that behavior because we've had client indicate that they cannot separate the processes for legal reasons. 

Closing. But if we have a really good use case why this is necessary, we can revisit.
